### ISSUE TITLE
[semantic-dom-diff] Should strip empty class attributes

### ISSUE THREAD

**Issue by bennypowers** [open-wc/open-wc#554]
`semantic-dom-diff` Should strip empty `class` attrs by default.

The only semantic meaning would be if someone puts a `[class] { /* ... */ }` rule in their css,
… but in my opinion, that person has it coming to them :wink:.

Empty `class` is a common symptom of `classMap`: You add some classMap to your code, but the case only obtains for one of your tests, everything still fails because of an empty `class` attribute which has otherwise no effect

Users can of course add `<div class>` to expected DOM, but this is 
1. a hassle
2. not semantic

Users could also use `{ ignoreAttributes: ['class'] }`, but this would strip *all* class attributes, where there might be meaningful static or dynamic class lists elsewhere in the tested HTML.

In short, `class=""` is *not semantic*, therefore `semantic-dom-diff` should strip it

@daKmoR suggested in slack a new `{ stripEmptyAttributes: ['class'] }` option with `class` in the defaults. Theoretically, all global, non-boolean attrs can go in by default. But in practise that probably only includes `class` and `id`.

**Comment by Westbrook** [comment]
Tangential question, is this an issue with `semantic-dom-diff` or with `classMap`? Seems like `classMap` would benefit from working more like `ifDefined`. Have looked into the `lit-html` teams thoughts on this?

**Comment by bennypowers** [comment]
I chatted with Justin about it on Polymer Slack. He didn't seem to fussed about it. I still think it'd be worth our attention here, since the package is after-all called **semantic**-dom-diff

**Comment by Westbrook** [comment]
Cool. Thanks for the background info. If they’re not looking into it, we definitely should!
